src/Subscription.ts

```diff
@@ -126,4 +119,28 @@ export class UnsubscriptionError extends Error {
     super('unsubscriptoin error(s)');
     this.name = 'UnsubscriptionError';
   }
+}
+
+function callPrivateUnsubscribe(sub: Subscription, result: { hasErrors: boolean; errors: any[]; }): void {
```

Comment:
This might perform better as a prototypal member. Since here it's external to the class, to use it in a class method you're forced to close over it, creating a closure.